dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserControllerUtils.java

```diff
@@ -118,11 +118,15 @@ public ObjectNode getUserDataApprovalWorkflows(User user) {
   private ArrayNode getWorkflowLevelNodes(User user, DataApprovalWorkflow workflow) {
     Set<String> authorities = user.getAllAuthorities();
 
-    boolean canApprove = authorities.contains(AUTHORITY_ALL) || authorities.contains(AUTH_APPROVE);
+    boolean canApprove =
+        authorities.contains(Authorities.ALL.toString())
+            || authorities.contains(F_APPROVE_DATA.toString());
     boolean canApproveLowerLevels =
-        authorities.contains(AUTHORITY_ALL) || authorities.contains(AUTH_APPROVE_LOWER_LEVELS);
+        authorities.contains(Authorities.ALL.toString())
+            || authorities.contains(F_APPROVE_DATA_LOWER_LEVELS.toString());
     boolean canAccept =
-        authorities.contains(AUTHORITY_ALL) || authorities.contains(AUTH_ACCEPT_LOWER_LEVELS);
+        authorities.contains(Authorities.ALL.toString())
+            || authorities.contains(F_ACCEPT_DATA_LOWER_LEVELS.toString());
```

Comment:
I feel like we should not get all authorities and make contains checks but instead to individual "isAuthorized" checks on `UserDetails` for the user.

Reply:
yep, that can be much simpler. I'll update.